wls-basisdaten-service/src/test/java/de/muenchen/oss/wahllokalsystem/basisdatenservice/rest/wahlbezirke/WahlbezirkeControllerIntegrationTest.java

```diff
@@ -0,0 +1,217 @@
+    WahlModelMapper wahlModelMapper;
+
+    @Autowired
+    WahlbezirkRepository wahlbezirkRepository;
+
+    @Autowired
+    WahltagRepository wahltagRepository;
+
+    @Autowired
+    WahlRepository wahlRepository;
+
+    @Autowired
+    ExceptionFactory exceptionFactory;
+
+    @AfterEach
+    void tearDown() {
+        SecurityUtils.runWith(Authorities.REPOSITORY_DELETE_WAHLBEZIRK, Authorities.REPOSITORY_DELETE_WAHL, Authorities.REPOSITORY_DELETE_WAHLTAG);
+        wahlbezirkRepository.deleteAll();
+        wahlRepository.deleteAll();
+        wahltagRepository.deleteAll();
+    }
+
+    @BeforeEach
+    void setup() {
+        WireMock.resetAllRequests();
+    }
+
+    @Nested
+    class GetWahlbezirke {
+
+        @Test
+        void loadedFromExternal() throws Exception {
+            val forWahltagDate = LocalDate.now().minusMonths(2);
+            val wahltagID = "_identifikatorWahltag1";
+            val wahltagNummer = "nummerWahltag1";
+            val wahlen = MockDataFactory.createWahlEntityList();
+            wahlRepository.saveAll(wahlen);
+
+            val eaiWahlbezirke = MockDataFactory.createSetOfClientWahlbezirkDTO(forWahltagDate);
+            WireMock.stubFor(WireMock.get("/wahldaten/wahlbezirk?forDate=" + forWahltagDate + "&withNummer=" + wahltagNummer)
+                    .willReturn(WireMock.aResponse().withHeader("Content-Type", "application/json").withStatus(HttpStatus.OK.value())
+                            .withBody(objectMapper.writeValueAsBytes(eaiWahlbezirke))));
+
+            val repoWahltage = MockDataFactory.createWahltagList("");
+            wahltagRepository.saveAll(repoWahltage);
+
+            val request = MockMvcRequestBuilders.get("/businessActions/wahlbezirke/" + wahltagID);
+
+            val responseFromController = api.perform(request).andExpect(status().isOk()).andReturn();
+            val responseBodyAsDTO = objectMapper.readValue(responseFromController.getResponse().getContentAsString(),
+                    de.muenchen.oss.wahllokalsystem.basisdatenservice.rest.wahlbezirke.WahlbezirkDTO[].class);
+
+            val expectedResponseBody = wahlbezirkDTOMapper
+                    .fromListOfWahlbezirkModelToListOfWahlbezirkDTO(
+
+                            wahlbezirkModelMapper.toWahlbezirkModelListMergedWithWahlenInfo(
+                                    wahlbezirkeClientMapper.fromRemoteSetOfDTOsToSetOfModels(eaiWahlbezirke),
+                                    wahlModelMapper.fromListOfWahlEntityToListOfWahlModel(wahlRepository.findByWahltagOrderByReihenfolge(forWahltagDate)),
+                                    exceptionFactory));
+
+            Assertions.assertThat(responseBodyAsDTO).containsExactlyInAnyOrderElementsOf(expectedResponseBody);
+        }
+
+        @Test
+        @Transactional
+        void externalDataIsPersisted() throws Exception {
+            val forWahltagDate = LocalDate.now().minusMonths(2);
+            val wahltagID = "_identifikatorWahltag1";
```

Comment:
ich denke es wäre es gut, wenn die Variablen gemäß dem namen des Datentypen angepasst werden und ebenfalls von identifikator in ID umbenannt werden.

Sonst hab ich nichts weiter, sieht gut aus 👍

Reply:
:) im Prinzip wäre ich deiner Meinung, aber genau das war meine Absicht, dem Betrachter eine Verknüpfung im Gehirn mit dem Namen dieser Property wie sie in der EAI heißt auszulösen. In der EAI heisst die iD des Wahltages und des Wahlbezirkes "identifikator".

Reply:
Wenn du das nicht akzeptierst ändere ich es.

Reply:
ich finde die Begründung ist reasonable.